core/src/main/java/hudson/model/Cause.java

```diff
@@ -453,9 +453,9 @@ public String getShortDescription() {
 
         @Override
         public void print(TaskListener listener) {
+
             listener.getLogger().println(Messages.Cause_UserIdCause_ShortDescription(
-                    // TODO JENKINS-48467 - better to use ModelHyperlinkNote.encodeTo(User), or User.getUrl, since it handles URL escaping
-                    ModelHyperlinkNote.encodeTo("/user/"+getUserIdOrUnknown(), getUserName())));
+                    ModelHyperlinkNote.encodeTo(User.getOrCreateByIdOrFullName(getUserIdOrUnknown()))));
```

Comment:
This will just produce a bogus hyperlink if the user is unknown: https://github.com/jenkinsci/jenkins/pull/3162#discussion_r155142633. We do not wish to create a hyperlink at all unless there is someone to link to.

Also we want to do lookup by ID only, never full name.